### PR TITLE
Update `overlap_mask` description.

### LINKED ISSUE #15684: Overlap mask when inference
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained a YOLOv8 model for instance segmentation. During inference, some objects have masks with missing middle parts, which overlap with other objects. For example, as shown in the figures below, Figure 1 represents the mask of a car's hood, while Figure 2 shows the mask of a specific component located on that hood. I would like the first mask to include the second mask. How can I achieve this?

First mask:
image

Second mask:
image

### Additional

_No response_

### LINKED ISSUE #17323: Overlaping masks not working on custom dataset
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I would like to ask, when I am training the yolov8 for instance segmentation task directly on roboflow, it gives me overlaping segmentation masks as expected. 
Untitled

However when I am training and testing yolov8n-seg model locally I am struggling to get overlapping masks even when I adjusted parameter overlap_mask=True along with iou and conf as I found in other related issues. Please can you help me with this? What else should I change to get overlapping masks? Code sample:
    model = YOLO('yolov8n-seg.pt')
    results = model.train(data=data_path,
                          project=project,
                          name=name,
                          epochs=200,
                          patience=0,  # Disable early stopping
                          batch=6,
                          imgsz=800,
                          overlap_mask=True,
                          boxes=False,
                          save=True,
                          show=True,
                          verbose=False,
                          conf=0.2,
                          iou=0.7,
                          mask_ratio=2) 
   metrics = model.val()
   results1 = model.predict(source='26.jpeg', conf=0.2, iou=0.6, boxes=False, overlap_mask=True)
   Example of tray output mask with wrong holes: 
image

### Additional

_No response_

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes https://github.com/ultralytics/ultralytics/issues/17323 Closes https://github.com/ultralytics/ultralytics/issues/15684

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Updated training documentation and configuration for handling segmentation masks to improve clarity and functionality.

### 📊 Key Changes
- **Documentation Update**: Clarified the behavior of the `overlap_mask` option in the training arguments documentation.
- **Configuration Comment**: Revised comments in the default configuration file to reflect changes about merging masks during training.

### 🎯 Purpose & Impact
- **Improved Understanding**: The updated documentation and comments ensure users clearly understand how segmentation masks are managed during training, specifying the merging process.
- **Enhanced Model Training**: By specifying the overlay behavior of smaller masks on larger ones, users can better configure training for specific segmentation needs, potentially leading to more accurate model outcomes.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting your pull request to the `ultralytics/ultralytics` repository! 🚀 This is an automated response, and an Ultralytics engineer will assist you soon. Let's make sure everything is set for a smooth review process:

- ✅ **Define a Purpose**: Your PR description should clearly explain the purpose of updating the `overlap_mask` description and link it to any [relevant issues](https://github.com/ultralytics/ultralytics/issues) like you've done with #17323 and #15684. 
- ✅ **Synchronize with Source**: Ensure your branch aligns with the `main` branch of the `ultralytics/ultralytics` repository. If necessary, update it by using the 'Update branch' button or execute `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Confirm all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are successful. If any checks fail, addressing them will be crucial so that your updates can be merged smoothly.
- ✅ **Update Documentation**: Verify that all relevant [documentation](https://docs.ultralytics.com) reflects your changes for clarity on how segmentation masks are managed during training.
- ✅ **Add Tests**: Ensure any related tests are up-to-date with your changes and pass successfully. This helps maintain the integrity of the codebase.
- ✅ **Sign the CLA**: If this is your first contribution to Ultralytics, please sign our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) with a message stating "I have read the CLA Document and I sign the CLA."
- ✅ **Minimize Changes**: Try to keep the alterations to the code as minimal as possible, focusing only on what's necessary for clarifying the `overlap_mask` parameter.

For detailed instructions and best practices, please follow our [Contributing Guide](https://docs.ultralytics.com/help/contributing). If you have further questions, feel free to ask. Thank you for enhancing our project! 😄

**Comment by glenn-jocher** [comment]
@Y-T-G good fix!
